frontend/src/libs/components/button/button.tsx

```diff
@@ -8,15 +8,21 @@ import styles from './styles.module.scss';
 type Properties = {
   label: string;
   type?: 'button' | 'submit';
-  iconName?: IconName;
-  iconColor?: ValueOf<typeof IconColor>;
+  iconName?: IconName | undefined;
+  iconColor?: ValueOf<typeof IconColor> | undefined;
   iconWidth?: number;
   iconHeight?: number;
-  style?: 'primary' | 'secondary' | 'rounded' | 'rounded-transparent' | 'add';
+  style?:
+    | 'primary'
+    | 'secondary'
+    | 'rounded'
+    | 'rounded-transparent'
+    | 'add'
+    | 'icon-right';
   isLoading?: boolean;
   isDisabled?: boolean;
   isLabelVisuallyHidden?: boolean;
-  onClick?: () => void;
+  onClick?: (() => void) | undefined | (() => Promise<void>);
```

Comment:
```suggestion
  onClick?: () => void | undefined;
```

Reply:
@Liza-Veis without it i can`t make useCallback(handleDelete) in delete journal modal async
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92929504/d2e34d44-6806-46e0-b878-08570f260764)

Reply:
Have you tried to add void?

Reply:
@Liza-Veis yep, and it does not help,